components/filter/filter.js

```diff
@@ -207,13 +214,22 @@ class Filter extends LocalizeCoreElement(RtlMixin(LitElement)) {
 		if (dimension.loading) {
 			return html`
 				<d2l-loading-spinner></d2l-loading-spinner>
-				<div class="d2l-offscreen">${this.localize('components.filter.loading')}</div>
+				<div class="d2l-offscreen" aria-busy="true" role="alert">${this.localize('components.filter.loading')}</div>
 			`;
 		}
 
+		if (this._isDimensionEmpty(dimension)) {
+			return html`
+                <div class="d2l-filter-dimension-info-message d2l-body-small" role="alert">
+                    ${this.localize('components.filter.noFilters')}
+                </div>
+            `;
+		}
+
 		return html`
 			<d2l-list
 				@d2l-list-selection-change="${this._handleChangeSetDimension}"
+				data-key="${dimension.key}"
```

Comment:
This simplifies the `_handleChangeSetDimension` logic, especially now that there's a `div` between the list and the `hierarchical-view`